mathesar/api/db/viewsets/tables.py

```diff
@@ -146,3 +148,28 @@ def previews(self, request, pk=None):
         )
 
         return Response(table_data)
+
+    @action(methods=['post'], detail=True)
+    def existing_import(self, request, pk=None):
+        temp_table = self.get_object()
+        serializer = TableImportSerializer(data=request.data, context={"request": request})
+        serializer.is_valid(raise_exception=True)
+        existing_table_id = serializer.validated_data['table_to_import_to']
+        data_files = serializer.validated_data['data-files']
+        mappings = serializer.validated_data['mappings']
+        existing_table = get_table_or_404(existing_table_id)
+
+        try:
+            insert_records_to_existing_table(
+                existing_table, temp_table, mappings, data_files
+            )
```

Comment:
@mathemancer  Should we make this a method under the `Table` model?

Reply:
Yes, I think that would be good.